src/history.ts

```diff
@@ -0,0 +1,45 @@
+import { randomUUID } from 'node:crypto';
+import { type HistoryIssue } from './shared';
+
+export const history = new Array<HistoryIssue>();
+
+export const addNewHistoryIssue = ({ action, issue, replacer }: Omit<HistoryIssue, 'id'>) => {
+	if (history.length >= 100) {
+		history.shift();
+	}
+
+	let original: string;
+
+	//  If action is Replace or ReplaceAll, the word should be replaced with the replacer
+	// on the issue object, because when the user rolls back the action and write a replacement
+	// again, the replacer should be the new word to replace the issue to avoid overflowing characters.
+	if (replacer) {
+		// Don't record the action if the replacer is the same as the issue text.
+		if (replacer === issue.text) {
```

Comment:
This is a good idea, could you add this check in display.ts too where it is like `if (!replacer) { return Action.Ignore };`

Reply:
You mean this? 😬
```ts
if (!replacer || replacer === issue.text) {
  return [Action.Ignore, ''];
}
```

Reply:
Yeah, I was thinking that if someone accidentally clicks the replace button, they might think, "oh, I'll just type in the same word so nothing changes".

Reply:
You have a point, even I made it once 🤣